shop/src/data/fetchProductStock.js

```diff
@@ -0,0 +1,19 @@
+import memoize from 'lodash/memoize'
+
+const fetchProductStock = memoize(
```

Comment:
Are we sure we want to cache this value given the inventory will change as buyers get the item?

Reply:
Without caching, there are too many requests to fetch the stock. That's the reason I added it.

Reply:
Leaving the `productId` undefined will do the same thing.